controllers/module_version_label_action_table.go

```diff
@@ -0,0 +1,66 @@
+package controllers
+
+import "github.com/kubernetes-sigs/kernel-module-management/internal/utils"
+
+const (
+	labelMissing   = "missing"
+	labelPresent   = "present"
+	labelDifferent = "different"
+
+	addAction    = "add"
+	deleteAction = "delete"
+	noneAction   = "none"
+)
+
+type labelActionKey struct {
+	module       string
+	moduleLoader string
+	devicePlugin string
+}
+
+type labelAction struct {
+	getLabelName func(string, string) string
+	action       string
+}
+
+var labelActionTable = map[labelActionKey]labelAction{
+	labelActionKey{
+		module:       labelMissing,
+		moduleLoader: labelMissing,
+		devicePlugin: labelMissing}: labelAction{getLabelName: nil, action: noneAction},
+
+	labelActionKey{
+		module:       labelMissing,
+		moduleLoader: labelPresent,
+		devicePlugin: labelPresent}: labelAction{getLabelName: utils.GetDevicePluginVersionLabelName, action: deleteAction},
+
+	labelActionKey{
+		module:       labelMissing,
+		moduleLoader: labelPresent,
+		devicePlugin: labelMissing}: labelAction{getLabelName: utils.GetModuleLoaderVersionLabelName, action: deleteAction},
+
+	labelActionKey{
+		module:       labelPresent,
+		moduleLoader: labelMissing,
+		devicePlugin: labelMissing}: labelAction{getLabelName: utils.GetModuleLoaderVersionLabelName, action: addAction},
+
+	labelActionKey{
+		module:       labelPresent,
+		moduleLoader: labelPresent,
+		devicePlugin: labelMissing}: labelAction{getLabelName: utils.GetDevicePluginVersionLabelName, action: addAction},
+
+	labelActionKey{
+		module:       labelPresent,
+		moduleLoader: labelPresent,
+		devicePlugin: labelPresent}: labelAction{getLabelName: nil, action: noneAction},
+
+	labelActionKey{
+		module:       labelPresent,
+		moduleLoader: labelDifferent,
+		devicePlugin: labelDifferent}: labelAction{getLabelName: utils.GetDevicePluginVersionLabelName, action: deleteAction},
+
+	labelActionKey{
+		module:       labelPresent,
+		moduleLoader: labelDifferent,
+		devicePlugin: labelMissing}: labelAction{getLabelName: utils.GetModuleLoaderVersionLabelName, action: deleteAction},
```

Comment:
It means that the user has overwritten the version label right?! So the new DS will be deploy in parallel with removing the old labels?

Reply:
yes, correct , user has set the new value to its version. So the action for this state is to just remove the label of the module-loader, which will cause the removal of the pod. new DS pod will not be deployed since we have not added the module-loader label with new version yet

Reply:
> new DS pod will not be deployed since we have not added the module-loader label with new version yet

And this will automatically be triggered by the action in index 4 in the table once the labels are deleted right?!

Reply:
yes, change in the module version labels triggers the reconciliation loop